verl/trainer/ppo/ray_trainer.py

```diff
@@ -419,10 +419,18 @@ def _validate_config(self):
         config = self.config
         # number of GPUs total
         n_gpus = config.trainer.n_gpus_per_node * config.trainer.nnodes
+        if config.actor_rollout_ref.actor.strategy == "megatron":
+            model_parallel_size = config.actor_rollout_ref.actor.megatron.tensor_model_parallel_size * config.actor_rollout_ref.actor.megatron.context_parallel_size * config.actor_rollout_ref.actor.megatron.pipeline_model_parallel_size
+            assert n_gpus % model_parallel_size == 0, f"n_gpus ({n_gpus}) must be divisible by model_parallel_size ({model_parallel_size})"
+            megatron_dp = n_gpus // model_parallel_size
+            assert megatron_dp % config.actor_rollout_ref.actor.megatron.expert_model_parallel_size == 0, f"data parallel size ({megatron_dp}) must be divisible by expert_model_parallel_size ({config.actor_rollout_ref.actor.megatron.expert_model_parallel_size})"
+            minimal_bsz = megatron_dp * config.actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu
```

Comment:
This line is not necessary

Reply:
We only need to make sure real_train_batch_size % dp == 0. In FSDP, dp_size = n_gpus. In megatron, dp_size = n_gpus // tp // pp // cp // ep (depends on how we use ep)